packages/eas-cli/src/commands/update/index.ts

```diff
@@ -584,6 +571,59 @@ export default class UpdatePublish extends EasCommand {
   }
 }
 
+export async function getUpdatesToRepublishInteractiveAsync(
+  projectId: string,
+  branchName: string,
+  platformFlag: string,
+  cumulativeUpdates: Exclude<
+    Exclude<ViewBranchUpdatesQuery['app'], null | undefined>['byId']['updateBranchByName'],
+    null | undefined
+  >['updates'] = [],
+  offset: number = 0
+): Promise<any> {
+  const fetchMoreValue = '_fetchMore';
+
+  const { updates } = await ensureBranchExistsAsync({
+    appId: projectId,
+    name: branchName,
+    offset,
+  });
```

Comment:
This is where I think we should pass in a page size. The code in charge of pagination and UX should determine the page size instead of ensureBranchExistsAsync.

This will also make it easier to fetch pageSize + 1 updates, where the extra update is a way to see if there are more to fetch. Right now there's a minor bug where if there are exactly a multiple of pageSize updates, we'll show "Next page..." even though there are no more pages.

Reply:
Great point about fetching an extra update to avoid that final page bug!

I'm not sure I fully understand the suggestion here (re page size) so I'll state what I think you're saying and let me know if it is inaccurate:
- page size (currently denoted as limit) should be explicitly defined here, rather using the global default of `getViewBranchUpdatesQueryUpdateLimit()`

Don't feel strongly either way, happy to make this change. I think we can avoid the bug you mentioned by just displaying `getViewBranchUpdatesQueryUpdateLimit() - 1` updates, rather than fetching a different number of updates altogether.

Reply:
I am saying the page size should always be passed in here rather than relying on the default. A page size of 50 or 100 might be more usable.

Separately, the number of updates we want to display per page should determine the number of updates to fetch rather than the other way around. It doesn't affect the server much to fetch 300 vs 301 updates. But displaying 299 updates per page is awkward.

Reply:
got it! i'll make the change shortly. thanks for clarifying.